heat/core/tests/test_suites/basic_test.py

```diff
@@ -0,0 +1,173 @@
+from unittest import TestCase
+from heat.core import dndarray, MPICommunication, MPI, types, factories
+import heat as ht
+
+import numpy as np
+import torch
+
+
+class BasicTest(TestCase):
+
+    __comm = MPICommunication()
+    __device = None
+
+    @property
+    def comm(self):
+        return BasicTest.__comm
+
+    @property
+    def device(self):
+        return BasicTest.__device
+
+    def get_rank(self):
+        return self.comm.rank
+
+    def get_size(self):
+        return self.comm.size
+
+    def assert_array_equal(self, heat_array: dndarray.DNDarray, numpy_array):
```

Comment:
i prefer to use torch for many things, can you add a torch_tensor here. and can you make it an option to choose between the two of them?